src/lib/mdns/Advertiser.h

```diff
@@ -28,6 +28,12 @@ namespace Mdns {
 
 static constexpr uint16_t kMdnsPort = 5353;
 
+enum class CommssionAdvertiseMode : uint8_t
+{
+    kCommissioning,
```

Comment:
how about adding a kUndefined and set that as a default, to force parameter builders to specifiy a mode?

uint8_t storage type is not required.